src/CoreMatchers/Descriptions/Description.cs

```diff
@@ -19,6 +19,16 @@ public IDescription AppendList(string before, string separator, string after, pa
             return appendList(before, separator, after, args);
         }
 
+        public IDescription AppendList(string before, string separator, string after, IEnumerable<IDescribable> args)
+        {
+            return appendList(before, separator, after, args);
+        }
+
+        public IDescription AppendValueList<T>(string start, string separator, string end, IEnumerable<T> values)
+        {
+            return AppendList(start, separator, end, new SelfDescribingValueEnumerable<T>(values));
```

Comment:
add a conditional check to ensure that `T` cannot already be treated as a `SelfDescribingValue`. If that is the case then we don't need to hit the existing constructor that wraps all the values in `SelfDescribingValue`. Create a new constructor for `SelfDescribingValueEnumerable` that accepts something like `IEnumerable<T> where T : SelfDescribingValue`. Disclaimer: I'm not sure about the syntax.